src/ConnectedMode/Binding/CompositeBindingConfigProvider.cs

```diff
@@ -20,31 +20,39 @@
 
 using System;
 using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Diagnostics;
 using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
 using SonarLint.VisualStudio.Core;
 using SonarLint.VisualStudio.Core.Binding;
+using SonarLint.VisualStudio.Integration;
+using SonarQube.Client;
 using SonarQube.Client.Models;
 
 namespace SonarLint.VisualStudio.ConnectedMode.Binding
 {
-    public class CompositeBindingConfigProvider : IBindingConfigProvider
+    /// <summary>
+    /// Composite that coordinates turning server Quality Profile information into a set of config files
+    /// for a particular language
+    /// </summary>
+    [Export(typeof(IBindingConfigProvider))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    internal class CompositeBindingConfigProvider : IBindingConfigProvider
     {
         private readonly HashSet<IBindingConfigProvider> providers;
 
-        public CompositeBindingConfigProvider(params IBindingConfigProvider[] providers)
-        {
-            // params args can't be null - will be an empty array
-            if (providers.Length == 0)
```

Comment:
The providers are no longer set in the public constructor, so I change the "throw" to an "assert".
There is a new unit test that checks that the public constructor creates the expected provider types.